pipelines/firebase/__init__.py

```diff
@@ -0,0 +1,84 @@
+"""Loads data from firebase realtime database"""
+
+from typing import Any, Iterator, List, Sequence
+
+import dlt
+from dlt.extract.source import DltResource
+from dlt.common.typing import TDataItem
+
+from firebase_admin import credentials, db
+
+
+@dlt.source(name="firebase")
+def firebase_source(
+    project_id: str =  dlt.secrets.value,
+    private_key_id: str =  dlt.secrets.value,
+    private_key: str = dlt.secrets.value,
+    client_email: str = dlt.secrets.value,
+    client_id: str = dlt.secrets.value,
+    database_url:  str = dlt.secrets.value
+) -> Sequence[DltResource]:
+    
+    def _get_data(credentials: Any, database_url: str) -> Iterator[TDataItem]:
+        """
+        Note: For this pipeline, we use a mock json template such below,
+        your firebase realtime data likely different:
+        {
+            "discography": [
+                {
+                    "albums": [
+                        {
+                            "album_name": str,
+                            "year": int
+                        }
+                    ],
+                    "band_name": str
+                }
+            ]
+        }
+        """
+
+        # initialize app
+        try:
+            firebase_admin.initialize_app(
+                credentials, 
+                {"databaseUrl": database_url}
+            )
+        except ValueError:
+            firebase_admin.get_app(name="[DEFAULT]")
+
+        ref = db.reference("/")
+        data = json.loads(ref.get())
```

Comment:
this is getting the whole database at once. any way to select or filter data? does the concept of incremental load apply here?

Reply:
OK! another question: do those objects contain unique id? does it depend on database schema?

Reply:
we can set the object name using `db.child("collection_name")`, otherwise it will be random